golink_test.go

```diff
@@ -4,17 +4,162 @@
 package golink
 
 import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
 	"testing"
 	"time"
 )
 
+func init() {
+	// tests always need golink to be run in dev mode
+	*dev = ":8080"
+}
+
+func TestServeGo(t *testing.T) {
+	var err error
+	db, err = NewSQLiteDB(":memory:")
+	if err != nil {
+		t.Fatal(err)
+	}
+	db.Save(&Link{Short: "who", Long: "http://who/"})
+	db.Save(&Link{Short: "me", Long: "/who/{{.User}}"})
+
+	tests := []struct {
+		name        string
+		link        string
+		currentUser func(*http.Request) (string, error)
+		wantStatus  int
+		wantLink    string
+	}{
+		{
+			name:       "simple link",
+			link:       "/who",
+			wantStatus: http.StatusFound,
+			wantLink:   "http://who/",
+		},
+		{
+			name:        "simple link, anonymous request",
+			link:        "/who",
+			currentUser: func(*http.Request) (string, error) { return "", nil },
+			wantStatus:  http.StatusFound,
+			wantLink:    "http://who/",
+		},
+		{
+			name:       "user link",
+			link:       "/me",
+			wantStatus: http.StatusFound,
+			wantLink:   "/who/foo@example.com",
+		},
```

Comment:
Does it make sense to add any negative tests here?

Reply:
done